icicle/appUtils/ntt/ntt.cu

```diff
@@ -501,11 +501,11 @@ namespace ntt {
       h_coset.clear();
     }
 
-    const bool is_small_ntt = logn < 16;                  // cutoff point where mixed-radix is faster than radix-2
+    // (heuristic) cutoff point where mixed-radix is faster than radix-2
+    const bool is_small_ntt = (logn < 16) && ((size_t)size * batch_size < (1 << 20));
```

Comment:
why not have a table? It's only 16 lines

Reply:
I will do that and later extend the table to account for orderings too.

Reply:
update: I will do it in the PR where adding orderings instead of doing it twice.